src/viewer.js

```diff
@@ -455,35 +455,32 @@ $.Viewer = function( options ) {
         $.console.warn('No valid drawers were selected. Using the default value.');
     }
     // extend the drawerOptions object with additional properties to pass to the Drawer implementation
-    this.drawer = null; // TO DO: how to deal with the possibility that none of the requested drawers are supported?
+    // TODO: how to deal with the possibility that none of the requested drawers are supported?
+    this.drawer = null;
     for(let i = 0; i < drawersToTry.length; i++){
-        let Drawer = drawersToTry[i];
-        let optsKey = null;
-        // replace text-based option with appropriate constructor
-        if (Drawer === 'canvas'){
-            Drawer = $.CanvasDrawer;
-            optsKey = 'canvas';
-        } else if (Drawer === 'html'){
-            Drawer = $.HTMLDrawer;
-            optsKey = 'html';
-        } else if (Drawer === 'webgl'){
-            Drawer = $.WebGLDrawer;
-            optsKey = 'webgl';
-        } else {
+
+        //todo necessary? why not to use class names as drawer IDs
+        let optsKey = drawersToTry[i];
+        let Drawer = $.determineDrawer(optsKey);
+        if (!Drawer) {
```

Comment:
I'd like to keep the ability to pass the constructor of a drawer in as an option.

Reply:
Fixed. I also removed filtering since it explicitly required drawer IDs. Now you can just implement new drawer on the openseadragon namespace and it gets recognized automatically. I removed also drawer checking and I use `instanceof` check instead.